cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -1174,13 +1197,42 @@ public MapAndStatus(T loadMap, K status) {
      */
     private KafkaRebalanceAnnotation rebalanceAnnotation(Reconciliation reconciliation, KafkaRebalance kafkaRebalance) {
         String rebalanceAnnotationValue = rawRebalanceAnnotation(kafkaRebalance);
+        KafkaRebalanceState state = state(kafkaRebalance);
         KafkaRebalanceAnnotation rebalanceAnnotation;
+
         try {
-            rebalanceAnnotation = rebalanceAnnotationValue == null ?
-                    KafkaRebalanceAnnotation.none : KafkaRebalanceAnnotation.valueOf(rebalanceAnnotationValue);
+            if (rebalanceAnnotationValue == null) {
+                rebalanceAnnotation = KafkaRebalanceAnnotation.none;
+            } else if (rebalanceAnnotationValue.equals("none")) {
+                rebalanceAnnotation = KafkaRebalanceAnnotation.unknown;
+            } else {
+                rebalanceAnnotation = KafkaRebalanceAnnotation.valueOf(rebalanceAnnotationValue);
+            }
+
+            if (rebalanceAnnotation != KafkaRebalanceAnnotation.none) {
```

Comment:
Why is this switch statement here? You already have code to exclude and highlight inappropriate annotation types in each of the state methods (`onProposalReady`, `onRebalancing` etc). This seems redundant? It also means you have maintain the mapping of what annotations are valid in each state in two places (this switch and in the state method). 

I suggest removing this switch statement and keeping the logic of what annotations are valid inside each state method.

Reply:
I guess we might require that since we are setting the rebalance annotation to `unknown` here

Reply:
Ok, after going through this with @ShubhamRwt he has convinced me this actually needed. However, I am still concerned that we have 2 locations to maintain valid annotations for each state. I wonder if it would be better, instead of doing the checking in the rebalanceAnnotations method and have that value given to `reconcile`, to have a generic `validateAnnotation` method that gets called from within each `onState` method. Something like:
```java
List<KafkaRebalanceAnnotations> validAnnotations = Arrays.asList(KafkaRebalanceAnnotation.stop);
KafkaRebalanceAnnotation rebalanceAnnotation = validateAnnotation("onRebalancing", validAnnotations);
```
That way, the list and logic for valid annotations for each state are kept only in the `onState` methods. 

WDYT @ppatierno? Am I over thinking this?

Reply:
I took a "static" look at the code and I think that what @tomncooper is saying should work. As an example ...
Imagine that someone set the `stop` annotation but we are in the `ProposalReady` state. This method would translate the `stop` into `unknown` but let's assume we remove the switch statement here and the annotation will remain `stop`. When the `onProposalReady` is executed, the `stop` will be handled in the `default` case of the switch and warning about the wrong annotation.
Even for the other `onXXXXX` method, any kind of not allowed annotation seems to be handled the same way by printing the warning.
Am I right? If yes, it could assume that the switch here can be removed as Tom's suggestion.

Reply:
@tomncooper @ppatierno I just have a small doubt. The generic methodwill check all if the rebalance annotation is correct or not as per the state, right ? But Iam not actually overriding the `rebalanceAnnotation` to `unknown` anywhere. So when I get back to the `reconcile` method, wouldn't this condition fail -->
```          if (rebalanceAnnotation.equals(KafkaRebalanceAnnotation.unknown)) {
                                                        return Future.succeededFuture();
                                                    } ```

Reply:
``` if (rebalanceAnnotation.equals(KafkaRebalanceAnnotation.unknown)) { 
            return Future.succeededFuture(); 
}```